cmd/root_test.go

```diff
@@ -19,157 +19,196 @@ package cmd
+		c.Fatal(err)
+	}
+	GitInitialHash = initHash.String()
+	head, _ := repo.Head()
+
+	{ // CWD is repo root
+		if err = os.Chdir(path); err != nil {
+			c.Fatal(err)
+		}
+		r, dir, err := hpcToolkitRepo()
+		c.Assert(err, IsNil)
+		c.Check(dir, Equals, path)
+		h, _ := r.Head()
+		c.Check(h.Hash(), Equals, head.Hash())
+	}
+
+	{ // CWD is subdir in repo root
+		subDir := filepath.Join(path, "subidr")
+		if err = os.MkdirAll(subDir, os.ModePerm); err != nil {
+			c.Fatal(err)
+		}
+		if err = os.Chdir(subDir); err != nil {
+			c.Fatal(err)
+		}
+		r, dir, err := hpcToolkitRepo()
+		c.Assert(err, IsNil)
+		c.Check(dir, Equals, subDir)
+		h, _ := r.Head()
+		c.Check(h.Hash(), Equals, head.Hash())
+	}
+
+	{ // CWD is root of sub repo in repo root
+		subRepo := filepath.Join(path, "subrepo")
+		if err = os.MkdirAll(subRepo, os.ModePerm); err != nil {
+			c.Fatal(err)
+		}
+		_, _, err = initTestRepo(subRepo)
+		if err != nil {
+			c.Fatal(err)
+		}
+		if err = os.Chdir(subRepo); err != nil {
+			c.Fatal(err)
+		}
+		r, dir, err := hpcToolkitRepo()
+		c.Assert(err, IsNil)
+		c.Check(dir, Equals, path)
+		h, _ := r.Head()
+		c.Check(h.Hash(), Equals, head.Hash())
+	}
+
+	{ // CWD is parent of repo root, hope it's not repo itself.
+		os.Chdir(filepath.Dir(path))
+		_, _, err = hpcToolkitRepo()
+		c.Check(err, Equals, git.ErrRepositoryNotExists)
+	}
 }
 
 func (s *MySuite) TestIsHpcToolkitRepo(c *C) {
-	// sub directory of an hpc-toolkit git repository
-	_, callDir, _, _ := runtime.Caller(0)
-	repoDir := filepath.Dir(filepath.Dir(callDir))
-	repo, _ := git.PlainOpen(repoDir)
-	got := isHpcToolkitRepo(*repo)
-	c.Assert(got, Equals, true)
-
-	// temporary git repo is not a hpc-toolkit repo
-	storer := memory.NewStorage()
-	fs := memfs.New()
-	testRepo, _ := git.Init(storer, fs)
-	initTestRepo(*testRepo, fs)
-	got = isHpcToolkitRepo(*testRepo)
-	c.Assert(got, Equals, false)
+	repo, initHash, err := initTestRepo(c.MkDir())
+	if err != nil {
+		c.Fatal(err)
+	}
+
+	// Doesn't match
+	GitInitialHash = randomGitHash
+	c.Check(isHpcToolkitRepo(*repo), Equals, false)
+
+	// Matches
+	GitInitialHash = initHash.String()
+	c.Check(isHpcToolkitRepo(*repo), Equals, true)
 }
 
 func (s *MySuite) TestCheckGitHashMismatch(c *C) {
-	_, callDir, _, _ := runtime.Caller(0)
-	repoDir := filepath.Dir(filepath.Dir(callDir))
-	workDir := filepath.Dir(callDir)
-	repo, _ := git.PlainOpen(repoDir)
+	path := c.MkDir()
+	repo, init, err := initTestRepo(path)
+	if err != nil {
+		c.Fatal(err)
+	}
+	GitInitialHash = init.String()
+	if err = os.Chdir(path); err != nil {
+		c.Fatal(err)
+	}
+
 	head, _ := repo.Head()
 	hash := head.Hash().String()
 	branch := head.Name().Short()
 
-	// verify current working directory git hash against hash of v1.0.0
-	mockInjectedGitVariables()
-	mismatch, b, h, dir := checkGitHashMismatch()
-	c.Assert(mismatch, Equals, true)
-	c.Assert(dir, Equals, workDir)
-	c.Assert(b, Equals, branch)
-	c.Assert(h, Equals, hash)
+	{ // Matches
+		GitCommitHash = hash
```

Comment:
For the first 3 of the 4 cases outlined here logic is depending on `GitCommitHash` and `GitInitialHash` as opposed to just setting the one needed for the case I think the tests would be more readable if you set both at the top of each case. So for example on the "Matches" case you would start with:
```
GitCommitHash = hash
GitInitialHash = init.String()
```